### PR TITLE
[Model Update]: market palce offer_v2.0.0

### LINKED ISSUE #482: [Model Update]: Market Place Offer v2.0.0
## Update Reason
add attributes: Error Codes, Part Marking, Vehicle Description, catenaXId, unit for price, image, attachment, serial number, price, tax percentage, dimensions (length, height, width, diameter, weight), productionDate, areaOfUsage.

reference document: https://confluence.catena-x.net/display/CM/Data+model+MarketplaceOffer_update
Current version: v1.4.0 https://github.com/eclipse-tractusx/sldt-semantic-models/tree/main/io.catenax.market_place_offer/1.4.0

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists
- [x] The potential updates are discussed with all stakeholders
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [x] The potential update will be either 
  - [ ] backward compatible or 
  - [x] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by yuchengluo1**
## Description
As mentioned in issue: https://github.com/eclipse-tractusx/sldt-semantic-models/issues/482.

 -->

Closes #482 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by yuchengluo1** [comment]
@agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?

**Comment by agg3fe** [comment]
> @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?

The solution would be to first convert PhysicalDimensions aspect model to samm meta model.

**Comment by yuchengluo1** [comment]
> > @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?
> 
> The solution would be to first convert PhysicalDimensions aspect model to samm meta model.



> @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?



> > @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?
> 
> The solution would be to first convert PhysicalDimensions aspect model to samm meta model.

Do I need to raise a new model version (v2.0.1) for it?
According to the record, the samm model was reverted back to bamm model last year:
[Revert "update to samm"](https://github.com/eclipse-tractusx/sldt-semantic-models/commit/78f7d6a327d09c31e714f2a7510af7318db48965)

**Comment by yuchengluo1** [comment]
> > @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?
> 
> The solution would be to first convert PhysicalDimensions aspect model to samm meta model.

Do I need to raise a new model version (v2.0.1) for it?
According to the record, the samm model was reverted back to bamm model last year:
[Revert "update to samm"](https://github.com/eclipse-tractusx/sldt-semantic-models/commit/78f7d6a327d09c31e714f2a7510af7318db48965)

**Comment by agg3fe** [comment]
> > > @agg3fe This updated model version contains a shared-aspect (the PhysicalDimensions aspect model), which is still modelled with old bamm-model version, which caused conflict when it is merged into a samm-model. What should I do in this case?
> > 
> > 
> > The solution would be to first convert PhysicalDimensions aspect model to samm meta model.
> 
> Do I need to raise a new model version (v2.0.1) for it? According to the record, the samm model was reverted back to bamm model last year: [Revert "update to samm"](https://github.com/eclipse-tractusx/sldt-semantic-models/commit/78f7d6a327d09c31e714f2a7510af7318db48965)

I am not sure about the reason for converting back to bamm. But a new version 3.0.0 would be required as this is not compatible with old 2.0.0 bamm version.

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 1]
```diff
@@ -0,0 +1,413 @@
+#######################################################################
+# Copyright (c) 2024 T-Systems International GmbH
```
Comment (catroest) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:2]: shouldn't this be 2022-2024?

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+:taxPercentage a samm:Property ;
+   samm:preferredName "tax percentage"@en ;
+   samm:description "Tax percentage applicable to the product"@en ;
+   samm:characteristic :Decimal ;
+   samm:exampleValue "10.5"^^xsd:decimal .
+
+:dimensions a samm:Property ;
+   samm:preferredName "dimensions"@en ;
+   samm:description "(length, height, width, diameter, weight)"@en ;
+   samm:characteristic ext-dimension:PhysicalDimensionsCharacteristic .
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:174]: please check if you are able to use `https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.shared.quantity/1.0.0/Quantity.ttl` model here.

Comment (yuchengluo1) [reply io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:174]: If it is linked to Quantity aspect, the property "dimensions" has both mass and linear characteristic. However I cannot link a single property to 2 characteristics. Do you have a solution here? Or should I still use the physical dimension aspect?

Comment (catroest) [reply io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:174]: Quantity: You can link it directly to each specific characterisitc (for reference i did the same in the digital product passport). For weight to the massCharacteristic, for the others each to the linear characteristic.
(Physical Dimensions: This model is very restrictive (as of now) and the unit can not be adjusted.)

Comment (agg3fe) [reply io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:174]: Thanks @catroest for helping here.
@yuchengluo1 here's the example https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.generic.digital_product_passport/3.0.0/DigitalProductPassport.ttl#L665.

**Comment by agg3fe** [comment]
@yuchengluo1 is this model ready for MS2 check?

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.market_place_offer:2.0.0#> .
+@prefix ext-part: <urn:samm:io.catenax.serial_part:2.0.0#> .
+@prefix ext-quantity: <urn:samm:io.catenax.shared.quantity:1.0.0#> .
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:33]: you can use quantity 2.0.0

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "Description of all needed information to place a market place offer on the marketplace."@en ;
+   samm:properties ( [ samm:property :sku; samm:optional true ] :quantity [ samm:property :condition; samm:optional true ] :pickupLocation [ samm:property :incoterms; samm:optional true ] :price [ samm:property :image; samm:optional true ] [ samm:property :attachment; samm:optional true ] :bundleOffer :marketplaceProduct [ samm:property :missingParts; samm:optional true ] [ samm:property :dismantled; samm:optional true ] [ samm:property :mechanicalDamage; samm:optional true ] [ samm:property :corroded; samm:optional true ] [ samm:property :discolored; samm:optional true ] [ samm:property :availabilityDate; samm:optional true ] [ samm:property :mileage; samm:optional true ] [ samm:property :burned; samm:optional true ] [ samm:property :errorCodes; samm:optional true ] [ samm:property :partMarking; samm:optional true ] [ samm:property :vehicleDescription; samm:optional true ] :serialNumber :taxPercentage :height :areaOfUsage :width :length :diameter :weight ext-part:date ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:sku a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:42]: `preferredName` missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "Stock Keeping Unit is an unique identifier of the dealer providing the product"@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "1002090, xYz.09, ABcXYZ" .
+
+:quantity a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:47]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "The available quantity of an offered product."@en ;
+   samm:characteristic ext-quantity:ItemQuantityCharacteristic .
+
+:condition a samm:Property ;
+   samm:description "The condition of the offered product."@en ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:52]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+   samm:characteristic :ConditionCharacteristic ;
+   samm:exampleValue "used" .
+
+:pickupLocation a samm:Property ;
+   samm:description "The pickup location for the offered item."@en ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:57]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+   samm:characteristic :GeographicalCoordinates .
+
+:incoterms a samm:Property ;
+   samm:description "The incoterms."@en ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:61]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+   samm:see <file:///C:/Users/YUL/AppData/Local/ASPECT-MODEL-EDITOR/https%2525253A%2525252F%2525252Ficcwbo.org%2525252Fpublication%2525252Fincoterms-2020-introduction%2525252F> ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "DAP (Delivered at Place)" .
+
+:price a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:66]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 10]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "The net price of the product after tax"@en ;
+   samm:characteristic :PriceInformation .
+
+:image a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:70]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 11]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "Images of the product."@en ;
+   samm:characteristic :ImageFileArray .
+
+:attachment a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:74]: preferredName missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 12]
```diff
@@ -0,0 +1,405 @@
...
+   samm:characteristic ext-quantity:MassCharacteristic .
+
+:ConditionCharacteristic a samm-c:Enumeration ;
+   samm:preferredName "Condition Characteristic"@en ;
+   samm:description "Condition of the article as enumeration."@en ;
+   samm:dataType xsd:string ;
+   samm-c:values ( "New" "Used" "Refurbished" "Other" ) .
+
+:GeographicalCoordinates a samm:Characteristic ;
+   samm:preferredName "geographical coordinates"@en ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:204]: `description` mising

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 13]
```diff
@@ -0,0 +1,405 @@
...
+   samm:see <file:///C:/Users/YUL/AppData/Local/ASPECT-MODEL-EDITOR/https%2525253A%2525252F%2525252Fen.wikipedia.org%2525252Fwiki%2525252FGeographic_coordinate_system> ;
+   samm:dataType :GeographicalCoordinate .
+
+:PriceInformation a samm:Characteristic ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:208]: description mising

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 14]
```diff
@@ -0,0 +1,405 @@
...
+   samm:preferredName "ImageFileArray"@en ;
+   samm:description "Array of image of the object."@en ;
+   samm:dataType :Array .
+
+:AttachmentArray a samm-c:Set ;
+   samm:preferredName "AttachmentArray"@en ;
+   samm:description "array of attachments"@en ;
+   samm:dataType :Array .
+
+:BundleOffer a samm:Characteristic ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:222]: description missing

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 15]
```diff
@@ -0,0 +1,405 @@
...
+
+:image a samm:Property ;
+   samm:description "Images of the product."@en ;
+   samm:characteristic :ImageFileArray .
+
+:attachment a samm:Property ;
+   samm:description "Attachments, such as CAD drawings or certificates"@en ;
+   samm:characteristic :AttachmentArray .
+
+:bundleOffer a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:78]: Property and the referenced Characteristic should not have the same name

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 16]
```diff
@@ -0,0 +1,405 @@
...
+   samm:description "The original manufacturer of a product."@en ;
+   samm:characteristic samm-c:Text ;
+   samm:exampleValue "BMW" .
+
+:productLink a samm:Property ;
+   samm:preferredName "product link"@en ;
+   samm:description "A link to either the product on the original marketplace/webshop or a link to the product specifications document."@en ;
+   samm:characteristic :ProductLinkCharacteristic .
+
+:technicalSpecification a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:336]: Property and the referenced Characteristic should not have the same name

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 17]
```diff
@@ -0,0 +1,405 @@
...
+:TechnicalSpecificationEntity a samm:Entity ;
+   samm:preferredName "technical specification"@en ;
+   samm:properties ( :key :technicalValue ) .
+
+:mainCategory a samm:Property ;
+   samm:preferredName "main category"@en ;
+   samm:characteristic :MainCategory ;
+   samm:exampleValue "Audio, video, navigation" .
+
+:subCategory a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:383]: Property and the referenced Characteristic should not have the same name

File: io.catenax.market_place_offer/RELEASE_NOTES.md [hunk 18]
```diff
@@ -1,6 +1,19 @@
 # Changelog
 All notable changes to this model will be documented in this file.
 
+## [2.0.0] - 2024-02-05
+### Added
+- new aspect based on shared aspect "recycling strategy certificate"
+- several other new aspects
+- link serveral more models
```
Comment (agg3fe) [inline io.catenax.market_place_offer/RELEASE_NOTES.md:8]: typo `serveral`

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

File: io.catenax.market_place_offer/RELEASE_NOTES.md (hunk 18 above)
Comment (yuchengluo1) [reply io.catenax.market_place_offer/RELEASE_NOTES.md:8]: deleted

Comment (agg3fe) [reply io.catenax.market_place_offer/RELEASE_NOTES.md:8]: link `serveral` more models... look closely :)

**Comment by yuchengluo1** [comment]
Does the model fullfill the MS2 requirement? Could we try to get MS3 approval this afternoon?

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 19]
```diff
@@ -0,0 +1,431 @@
...
+   samm:exampleValue "used" .
+
+:pickupLocation a samm:Property ;
+   samm:preferredName "pickup location"@en ;
+   samm:description "The pickup location for the offered item."@en ;
+   samm:characteristic :GeographicalCoordinates .
+
+:incoterms a samm:Property ;
+   samm:preferredName "Incoterms"@en ;
+   samm:description "The incoterms."@en ;
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:66]: Would be great if you can elaborate the description a bit more.

**Comment by agg3fe** [comment]
> Does the model fullfill the MS2 requirement? Could we try to get MS3 approval this afternoon?

Thanks for resolving the issues. Please resolve the two comments left, then we are good to go for MS3.

**Review by agg3fe — APPROVED** [review]
LGTM

File: io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl [hunk 20]
```diff
@@ -0,0 +1,431 @@
...
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#> .
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#> .
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix : <urn:samm:io.catenax.market_place_offer:2.0.0#> .
+@prefix ext-part: <urn:samm:io.catenax.serial_part:2.0.0#> .
```
Comment (agg3fe) [inline io.catenax.market_place_offer/2.0.0/MarketplaceOffer.ttl:32]: serial part 3.0.0

**Pull request merged**